Expand ComputeFiniteStrain docs

sapitts opened the pull request:
Includes more detail on the derivation using the stretch tensor to find the stretching rate tensor through the right Cauchy-Green deformation tensor


Refs #10516

sapitts committed:
Expand discussion of solving for D, Refs #10516

sapitts commented:
ping @bwspenc for review; just version 1 here

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/248511/) on f9a984f wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/12384/site)

This comment will be updated on new commits.

permcody commented:
@bwspenc - feel free to review and approve - just don't merge until we have our libMesh update in today (~10/24 15:00)

bwspenc commented on modules/tensor_mechanics/doc/content/source/materials/ComputeFiniteStrain.md:
```diff
@@ -5,23 +5,74 @@
+
+### Deformation Gradient
+
+The incremental motion over the time increment is assumed to be the inverse of
```
It seems like we should define \hat{F} first before talking about its inverse.

bwspenc commented on modules/tensor_mechanics/doc/content/source/materials/ComputeFiniteStrain.md:
```diff
@@ -5,23 +5,74 @@
+!alert-end!
+
+As shown in [cite:zhang2018modified], the incremental deformation gradient can
+be multiplicatly decomposed into an incremental rotation tensor, $\boldsymbol{\hat{R}}$,
```
```suggestion
be multiplicatively decomposed into an incremental rotation tensor, $\boldsymbol{\hat{R}}$,
```

bwspenc commented on modules/tensor_mechanics/doc/content/source/materials/ComputeFiniteStrain.md:
```diff
@@ -5,23 +5,74 @@
+
+As shown in [cite:zhang2018modified], the incremental deformation gradient can
+be multiplicatly decomposed into an incremental rotation tensor, $\boldsymbol{\hat{R}}$,
+and the incremental right stretch tensor, $\boldsymbol{\hat{U}}
```
```suggestion
and the incremental right stretch tensor, $\boldsymbol{\hat{U}}$
```

bwspenc commented on modules/tensor_mechanics/doc/content/source/materials/ComputeFiniteStrain.md:
```diff
@@ -5,23 +5,74 @@
 \boldsymbol{F}_n^{-1}$, where $\boldsymbol{F}_n$ is the total deformation gradient at time $t_n$.
+!alert-end!
+
+As shown in [cite:zhang2018modified], the incremental deformation gradient can
```
```suggestion
Following the explanation of this procedure given by [cite:zhang2018modified], the incremental deformation gradient can
```

bwspenc replied:
I didn't want to make it sound like we invented it!

bwspenc commented on modules/tensor_mechanics/doc/content/source/materials/ComputeFiniteStrain.md:
```diff
@@ -5,23 +5,74 @@
 ## Description
 
 The finite strain mechanics approach used in the MOOSE tensor_mechanics module is the incremental
```
```suggestion
This class is used to compute the strain increment, total strain, and incremental rotation for finite strain problems. The finite strain approach used is the incremental
```

bwspenc commented on modules/tensor_mechanics/doc/content/source/materials/ComputeFiniteStrain.md:
```diff
@@ -5,23 +5,74 @@
 The finite strain mechanics approach used in the MOOSE tensor_mechanics module is the incremental
-corotational form from
-[Rashid 1993](http://onlinelibrary.wiley.com/doi/10.1002/nme.1620362302/abstract).
+corotational form [citep:rashid1993incremental].
```
```suggestion
corotational form [citep:rashid1993incremental]. This approach computes logarithmic strains and strain increments.
```

bwspenc commented on modules/tensor_mechanics/doc/content/source/materials/ComputeFiniteStrain.md:
```diff
@@ -5,23 +5,74 @@
+\end{equation}
+
+This incremental streteching rate tensor can then be used as the work conjugate
+for a stress measure, as shown in [cite:zhang2018modified], or used to find a strain
```
```suggestion
for a stress measure, or used to compute another strain
```

bwspenc commented on modules/tensor_mechanics/doc/content/source/materials/ComputeFiniteStrain.md:
```diff
@@ -5,23 +5,74 @@
+
+This incremental streteching rate tensor can then be used as the work conjugate
+for a stress measure, as shown in [cite:zhang2018modified], or used to find a strain
+measure. Therefore `ComputeFiniteStrain` includes two different methods to solve
```
```suggestion
measure. The most computationally expensive part of this procedure is the spectral decomposition of $\boldsymbol{\hat{C}}$ to find $\boldsymbol{\hat{U}}$. This can be computed exactly using an [eigensolution](#eigensolution), but an approximation of this can be computed with much lower computational expense using a [Taylor expansion](#taylorexpansion) procedure. This class provides options to perform this calculation either way, but the Taylor expansion is the default.
```

bwspenc replied:
Delete the two lines below if you like that text.

bwspenc commented on modules/tensor_mechanics/doc/content/source/materials/ComputeFiniteStrain.md:
```diff
@@ -32,25 +83,13 @@ For this form, we assume
-According to [Rashid 1993](http://onlinelibrary.wiley.com/doi/10.1002/nme.1620362302/abstract), the
-stretching rate tensor $\boldsymbol{D}$ and rotation matrix $\boldsymbol{R}$ can be approximated using Taylor
-expansion as:
+The stretching rate tensor $\boldsymbol{D}$ and rotation matrix $\boldsymbol{R}$
```
```suggestion
The stretching rate tensor $\boldsymbol{D}$ and incremental rotation matrix $\hat{\boldsymbol{R}}$
```

bwspenc commented on modules/tensor_mechanics/doc/content/source/materials/ComputeFiniteStrain.md:
```diff
@@ -5,23 +5,74 @@
+for the stretching rate tensor: the [Taylor Expansion](#taylorexpansion) and the
+[Eigenstrain Decomposition](#eigensolution). The Taylor Expansion is the default.
+
+### Assumptions for the Incremental Deformation Gradient Formulation
```
This section feels a little out of place. Maybe we should just combine this stuff with the previous section.

sapitts replied:
okay. I remember we had talked about moving it, but I couldn't remember where to move the section

sapitts replied:
Note to self: add a sentence about consistency in the kernel with `use_displaced_mesh` below the finite strain example block

sapitts committed:
Add links to consistency btw stress and strain

discussion to address comments, Refs #10516

sapitts commented:
When I fixed up the commits from your changes @bwspenc I went a little overboard and accidentally combined all the commits into my original branch and removed your authorship -- sorry about that :( The changes that you made are all included in this version though

bwspenc commented on modules/tensor_mechanics/doc/content/source/materials/ComputeFiniteStrain.md:
```diff
@@ -4,53 +4,89 @@
+
+### Deformation Gradient Definition
+
+The deformation gradient represent the change in a material element from the
```
```suggestion
The deformation gradient represents the change in a material element from the
```

bwspenc commented on modules/tensor_mechanics/doc/content/source/materials/ComputeFiniteStrain.md:
```diff
@@ -4,53 +4,89 @@
+of the deformation graidient is used to find the incremental rotation tensor
+$\boldsymbol{\hat{R}}$ and the stretching rate $\boldsymbol{D}$.
+Following [cite:rashid1993incremental], the stretching rate tensor can be expressed in terms
+of 'incremental' right Cauchy-Green deformation tensor
```
```suggestion
of the 'incremental' right Cauchy-Green deformation tensor
```

bwspenc reviewed (commented):
Looks good.  I just made a couple really small suggestions.

bwspenc committed:
Fix grammar in docs, Refs #10516

Co-Authored-By: sapitts <stephanie.pitts@inl.gov>
